baselines/pFedHN/pFedHN/models.py

```diff
@@ -0,0 +1,222 @@
+            )
+            self.l1_bias = nn.Linear(hidden_dim, 120)
+            self.l2_weights = nn.Linear(hidden_dim, 84 * 120)
+            self.l2_bias = nn.Linear(hidden_dim, 84)
+            self.l3_weights = nn.Linear(hidden_dim, self.out_dim * 84)
+            self.l3_bias = nn.Linear(hidden_dim, self.out_dim)
+
+        if spec_norm and self.in_channels == 3:
+            self.c1_weights = spectral_norm(self.c1_weights)
+            self.c1_bias = spectral_norm(self.c1_bias)
+            self.c2_weights = spectral_norm(self.c2_weights)
+            self.c2_bias = spectral_norm(self.c2_bias)
+            self.l1_weights = spectral_norm(self.l1_weights)
+            self.l1_bias = spectral_norm(self.l1_bias)
+            self.l2_weights = spectral_norm(self.l2_weights)
+            self.l2_bias = spectral_norm(self.l2_bias)
+            self.l3_weights = spectral_norm(self.l3_weights)
+            self.l3_bias = spectral_norm(self.l3_bias)
+
+        elif spec_norm and self.in_channels == 1:
+            self.c1_weights = spectral_norm(self.c1_weights)
+            self.c1_bias = spectral_norm(self.c1_bias)
+            self.c2_weights = spectral_norm(self.c2_weights)
+            self.c2_bias = spectral_norm(self.c2_bias)
+            self.c3_weights = spectral_norm(self.c3_weights)
+            self.c3_bias = spectral_norm(self.c3_bias)
+            self.l1_weights = spectral_norm(self.l1_weights)
+            self.l1_bias = spectral_norm(self.l1_bias)
+            self.l2_weights = spectral_norm(self.l2_weights)
+            self.l2_bias = spectral_norm(self.l2_bias)
+            self.l3_weights = spectral_norm(self.l3_weights)
+            self.l3_bias = spectral_norm(self.l3_bias)
+
+    def forward(self, idx):
+        """Forward pass of the hypernetwork."""
+        emd = self.embeddings(idx)
+        features = self.mlp(emd)
+
+        if self.in_channels == 3:
+            weights = OrderedDict(
+                {
+                    "conv1.weight": self.c1_weights(features).view(
+                        self.n_kernels, self.in_channels, 5, 5
+                    ),
+                    "conv1.bias": self.c1_bias(features).view(-1),
+                    "conv2.weight": self.c2_weights(features).view(
+                        2 * self.n_kernels, self.n_kernels, 5, 5
+                    ),
+                    "conv2.bias": self.c2_bias(features).view(-1),
+                    "fc1.weight": self.l1_weights(features).view(
+                        120, 2 * self.n_kernels * 5 * 5
+                    ),
+                    "fc1.bias": self.l1_bias(features).view(-1),
+                    "fc2.weight": self.l2_weights(features).view(84, 120),
+                    "fc2.bias": self.l2_bias(features).view(-1),
+                    "fc3.weight": self.l3_weights(features).view(self.out_dim, 84),
+                    "fc3.bias": self.l3_bias(features).view(-1),
+                }
+            )
+
+        elif self.in_channels == 1:
```

Comment:
I presume this is for MNIST? maybe worth stating this briefly. If this (and the above) models are described in some section in the paper in greater detail, maybe add a reference to it.?

Reply:
sure.

Reply:
Description added